File: backend/routers/tax.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from database import get_db
from auth import get_current_user
import sqlite3
import os
from datetime import datetime

router = APIRouter()

DB_PATH = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))), "database", "safaripos.db")

@router.get("/summary")
async def tax_summary(current_user=Depends(get_current_user), db: Session = Depends(get_db)):
    if current_user.role not in ["admin", "manager"]:
        raise HTTPException(status_code=403, detail="Not authorized")
    
    conn = sqlite3.connect(DB_PATH)
    cursor = conn.cursor()
    
    # Today's tax
    today = datetime.now().strftime("%Y-%m-%d")
    cursor.execute("SELECT COALESCE(SUM(tax_amount), 0) FROM tax_ledger WHERE created_at LIKE ?", (today + "%",))
    today_tax = cursor.fetchone()[0]
    
    # This month
    month = datetime.now().strftime("%Y-%m")
    cursor.execute("SELECT COALESCE(SUM(tax_amount), 0) FROM tax_ledger WHERE created_at LIKE ?", (month + "%",))
    month_tax = cursor.fetchone()[0]
    
    # This year
    year = datetime.now().strftime("%Y")
    cursor.execute("SELECT COALESCE(SUM(tax_amount), 0) FROM tax_ledger WHERE created_at LIKE ?", (year + "%",))
    year_tax = cursor.fetchone()[0]
    
    # Total
    cursor.execute("SELECT COALESCE(SUM(tax_amount), 0) FROM tax_ledger")
    total_tax = cursor.fetchone()[0]
    
    conn.close()
    
    return {
        "today_tax": round(today_tax, 2),
        "month_tax": round(month_tax, 2),
        "year_tax": round(year_tax, 2),
        "total_tax": round(total_tax, 2)
    }

@router.get("/transactions")
async def tax_transactions(current_user=Depends(get_current_user), db: Session = Depends(get_db)):
    if current_user.role not in ["admin", "manager"]:
        raise HTTPException(status_code=403, detail="Not authorized")
    
    conn = sqlite3.connect(DB_PATH)
    cursor = conn.cursor()
    cursor.execute("SELECT receipt_no, created_at, product_name, quantity, unit_price, tax_rate, tax_amount, payment_method, cashier FROM tax_ledger ORDER BY id DESC LIMIT 100")
    rows = cursor.fetchall()
    conn.close()
    
    return [
        {
            "receipt_no": r[0],
            "date": r[1],
            "product": r[2],
            "quantity": r[3],
            "unit_price": r[4],
            "tax_rate": r[5],
            "tax_amount": r[6],
            "payment": r[7],
            "cashier": r[8]
        }
        for r in rows
    ]

File: frontend/js/tax.js
async function loadTaxReport() {
    try {
        const summary = await apiCall('/tax/summary');
        document.getElementById('todayTax').textContent = 'KSh ' + summary.today_tax.toFixed(2);
        document.getElementById('monthTax').textContent = 'KSh ' + summary.month_tax.toFixed(2);
        document.getElementById('yearTax').textContent = 'KSh ' + summary.year_tax.toFixed(2);
        document.getElementById('totalTax').textContent = 'KSh ' + summary.total_tax.toFixed(2);
        
        const transactions = await apiCall('/tax/transactions');
        const tbody = document.getElementById('taxTableBody');
        if (tbody) {
            tbody.innerHTML = transactions.map(t => 
                '<tr><td>' + t.receipt_no + '</td><td>' + t.date + '</td><td>' + t.product + '</td><td>' + t.quantity + '</td><td>' + t.tax_rate + '%</td><td>KSh ' + t.tax_amount.toFixed(2) + '</td><td>' + t.payment.toUpperCase() + '</td><td>' + t.cashier + '</td></tr>'
            ).join('') || '<tr><td colspan="8">No tax records yet</td></tr>';
        }
    } catch (e) { console.error(e); }
}
